### DISCUSSION TITLE
all traces of slip planes for FCC alloy

### DISCUSSION THREAD

**Discussion by LuqingCui** [mtex-toolbox/mtex#532] [Ask Anything] [answered]
Hi MTEX users,

I have the EBSD data of an FCC metal. I want to get the traces of different {111} slip planes.
Now I have got the traces, but I don't know they belong to which {111} slip plane.
**Can anyone help to distinguish different {111} planes with different colors and types of lines? such as (111) plane with a solid red line, and (1-11) with a yellow dotted line.**
The code and traces of slip planes are as follows:

```matlab
import_wizard
ebsd=ebsd('fe')
% reconstruct grains
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',15*degree);
% remove small grains
ebsd(grains(grains.diameter<=5)) = [];
% redo grain reconstruction
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',15*degree);
% smooth grain boundaries
grains = smooth(grains,5);
sS = slipSystem.fcc(ebsd.CS)
sS = sS.symmetrise;
plot(grains,grains.meanOrientation,'micronbar','off')
hold on
plot(grains.boundary,'linewidth',2)
hold off
hold on
for i = 1:length(sS)
sSlocal= grains.meanOrientation*sS(i);
quiver(grains,sSlocal.trace)
hold on
end
hold off
```
Traces of slip planes:
![Traces of slip planes](https://user-images.githubusercontent.com/72967420/96229309-fe3f3b80-0f96-11eb-8950-c1344f68488c.png)

Best regards,
Luqing Cui

**Comment by kilir** [comment, marked as answer]
Hi,
how about this (there are only 4 traces, so there's no need to go through all 24 sS):
```matlab
linestyles = {'-' '--' ':' '-.'}
colors = {'darkblue' 'fuchsia' 'purple' 'coral' }

h=symmetrise(Miller(1,1,1,cs),'unique','antipodal');
n111 = grains.meanOrientation*h;
n111.antipodal=1;

plot(grains,grains.meanOrientation,'micronbar','off','linewidth',2)
for i = 1:length(h)
hold on
quiver(grains,cross(n111(:,i),zvector),'linestyle',linestyles{i},'color',colors{i})
end
hold off
```
Cheers,
Rüdiger

**Reply by LuqingCui** [reply]
Hi Rüdiger，

Thanks for the help.
But a new question come out. When I run the code until "h=symmetrise(Miller(1,1,1,cs),'unique','antipodal');", Matlab give me a information of "Undefined function or variable 'cs'." See below:
![image](https://user-images.githubusercontent.com/72967420/96250084-94329080-0fae-11eb-885a-96db72f26123.png)

I add a code as "cs = crystalSymmetry('m-3m');" before the code "h=symmetrise(Miller(1,1,1,cs),'unique','antipodal');", it gives a information as:
![image](https://user-images.githubusercontent.com/72967420/96250485-33f01e80-0faf-11eb-8322-b4edebfc3ba0.png)

Do you know what went wrong?

Best regards,
Luqing Cui

**Reply by ralfHielscher** [reply]
you should replace `cs` by `ebsd.CS`

Ralf.

**Reply by LuqingCui** [reply]
Hi Ralf,

Great! It works!
Thanks for the help!

Best regards,
Luqing Cui

**Reply by Mustafa-Rifat** [reply]
Hi Ralf/Rüdiger,

Thanks a lot for this code. I am currently working on the same thing. I am working with FCC material and want to plot the trace of {111} slip planes. I have applied the code given by Rüdiger above, I am getting length(h)=8 i.e. my h is 8*1 Miller. However in the code we have defined 4 colors and linestyles. So when I run the loop I get this error (slip_plane_tace_all_v2 is my code name):
Index exceeds matrix dimensions.

Error in slip_plane_tace_all_v2 (line 52)
`quiver(grains,cross(n111(:,i),zvector),'linestyle',linestyles{i},'color',colors{i})`
I get a plot anyway:

![untitled](https://user-images.githubusercontent.com/60717121/103583305-83a6fd80-4ead-11eb-9dd9-899ef8b60b64.jpg)

To solve this issue: I introduced 4 more linestyles and colors and ran the code. Then it ran without any error. The modifications I did:
`linestyles = {'-' '--' ':' '-.' 'none' 'none' 'none' 'none'}`
`colors = {'r' 'g' 'b' 'c' 'm' 'y' 'k' 'w'}`
The new image I found:
![untitled2](https://user-images.githubusercontent.com/60717121/103583716-41ca8700-4eae-11eb-8197-c497c78892f3.jpg)
So, there is no change. Can you tell me why is this happening? is it alright?
Thanks a lot!

**Reply by Mustafa-Rifat** [reply]
Hi Rüdiger，

I am having a little bit difficulty in understanding this line. n111 in your code is in specimen coordinate, right? Why do we take the cross product of the n111's with z vector?
`quiver(grains,cross(n111(:,i),zvector),'linestyle',linestyles{i},'color',colors{i})`

Regards

**Reply by ralfHielscher** [reply]
Because a direction that is orthogonal to both the measurement surface and the plane normal gives us the trace. This is what a cross product computes.

Ralf.

**Reply by jucego333** [reply]
Dear @kilir , I also have the same issue "Index exceeds the number of array elements " How can we solve this problem? or does not it matter?

Best regards

**Reply by kilir** [reply]
Hi,
if you look at the example, there is a loop which does something , counting from 1 the length of h. If now you use this counter i of the loop to select something from a list, this list should obviously be at least as long as h.
Hope that helps. Have a look at what Mustafa did above.
Cheers,
Rüdiger

**Comment by Debo-rah** [comment]
Hi all,
i have a short question to the traces plotted with the code above. Are the plotted traces the planes or the plane normals? I need to plot the plane normals in my code...
thanks for helping!
Deborah
